cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -819,6 +822,32 @@ private void emitRunTimeMetric(ProgramRunId programRunId, ProgramRunStatus progr
     metricsContext.gauge(Constants.Metrics.Program.RUN_TIME_SECONDS, runTime);
   }
 
+  /**
+   * Emit stopping latency metric.
+   */
+  private void emitStoppingTimeMetric(ProgramRunId programRunId, ProfileId profileId, RunRecordDetail runRecord) {
+    if (runRecord.getStopTs() == null || runRecord.getStoppingTs() == null) {
+      return;
+    }
+    Map<String, String> additionalTags = new HashMap<>();
+    // don't want to add the tag if it is not present otherwise it will result in NPE
+    additionalTags.computeIfAbsent(Constants.Metrics.Tag.PROVISIONER,
+                                   provisioner -> SystemArguments.getProfileProvisioner(
+                                     runRecord.getSystemArgs()));
+    Map<String, String> tags = ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.PROFILE_SCOPE, profileId.getScope().name())
+      .put(Constants.Metrics.Tag.PROFILE, profileId.getProfile())
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
```

Comment:
looks like namespace and run id are already set by ProgramRunners.createProgramMetricsContext